### PR TITLE
Recovery services vault

### PR DESCRIPTION
**Pull request by arjenhuitema**
### File changes:

- Removed: src\resources\Microsoft.Authorization\policyDefinitions\deploy-rv_backuphealth_alert.bicep
- Removed: src\resources\Microsoft.Authorization\policyDefinitions\deploy-rv_backuphealth_alert.json
- Added: src\resources\Microsoft.Authorization\policyDefinitions\deploy-rv_backuphealth_monitor.bicep

### Changes to infra-as-code\bicep\deploy_dine_policies.bicep:

- Removed deployment: module rv_backuphealth_policy '../../src/resources/Microsoft.Authorization/policyDefinitions/deploy-rv_backuphealth_alert.bicep'
- Added deployment: module rv_backuphealthmonitor_policy '../../src/resources/Microsoft.Authorization/policyDefinitions/deploy-rv_backuphealth_monitor.bicep'

### Changes to src\resources\Microsoft.Authorization\policySetDefinitions\ALZ-MonitorLandingZone.json:
Removed parameters:

- RVBackupHealthAlertSeverity
- RVBackupHealthWindowSize
- RVBackupHealthEvaluationFrequency
- RVBackupHealthPolicyEffect
- RVBackupHealthAlertState
- RVBackupHealthThreshold

Added Parameters

- RVBackupHealthMonitorPolicyEffect

Other:

- Removed PolicyDefinition: "policyDefinitionId": "[concat('/providers/Microsoft.Management/managementGroups/',managementGroup().name, '/providers/Microsoft.Authorization/policyDefinitions/Deploy_RecoveryVault_BackupHealth_Alert')]"
- Added PolicyDefinition: "policyDefinitionId": "[concat('/providers/Microsoft.Management/managementGroups/',managementGroup().name, '/providers/Microsoft.Authorization/policyDefinitions/Deploy_RecoveryVault_BackupHealthMonitor_Alert')]"

### Changes to: infra-as-code\bicep\parameters-complete-landingzones.json:
Removed parameters: 

- RVBackupHealthAlertSeverity
- RVBackupHealthWindowSize
- RVBackupHealthEvaluationFrequency
- RVBackupHealthPolicyEffect
- RVBackupHealthAlertState
- RVBackupHealthThreshold

Added parameters:

- RVBackupHealthMonitorPolicyEffect

### Changes to: docs\wiki\PolicyInitiatives.md:

- Updated LZ table, removing old definition and including the new RSV alert definition.

### Evidence

Policy deployment

image

Remediation

image

Vault configuration
image

### CODE REVIEW

File: docs/wiki/Monitoring-and-Alerting-in-ALZ.md [hunk 1]
```diff
@@ -82,3 +80,34 @@ Available filters:
 - Signal type
 
 As an example, we could implement a filter on Severity (Critical, Error, Warning) only, ignoring (Informational, Verbose).
+
+## Monitoring Backup (Recovery Services Vaults)
+
+Azure Backup now provides new and improved alerting capabilities via Azure Monitor. The following policy: [Backup Monitor Policy](../blob/main/src/resources/Microsoft.Authorization/policyDefinitions/deploy-rv_backuphealth_monitor.bicep) configures new and existing recovery services vaults through a modify effect, which disables the classics alerts and enables the new built-in alerts.
```
Comment (jfaurskov) [inline docs/wiki/Monitoring-and-Alerting-in-ALZ.md:86]:
```suggestion
Azure Backup now provides new and improved alerting capabilities via Azure Monitor. The following policy: [Backup Monitor Policy](../blob/main/src/resources/Microsoft.Authorization/policyDefinitions/deploy-rv_backuphealth_monitor.bicep) configures new and existing recovery services vaults through a modify effect, which disables the classic alerts and enables the new built-in alerts. 
```

File: docs/wiki/Monitoring-and-Alerting-in-ALZ.md [hunk 2]
```diff
@@ -82,3 +80,34 @@ Available filters:
...
+                field: 'Microsoft.RecoveryServices/vaults/monitoringSettings.azureMonitorAlertSettings.alertsForAllJobFailures'
+                value: 'Enabled'
+            }
+        ]
+    }
+}
+```
+
+### Notifications
+While alerts are generated by default and can't be turned off for destructive operations, the notifications are in the control of the user, allowing you to clearly specify which set of email address (or other notification endpoints) you wish to route alerts to. Notifications are configured by an alert processing rule, which will be created by default when deploying the [ServiceHealth Initiative](../blob/main/src/resources/Microsoft.Authorization/policySetDefinitions/ALZ-MonitorServiceHealth.json)
```
Comment (jfaurskov) [inline docs/wiki/Monitoring-and-Alerting-in-ALZ.md:113]: Not sure what you mean by destructive operations?

File: src/resources/Microsoft.Authorization/policyDefinitions/deploy-rv_backuphealth_monitor.bicep [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+module BackupHealthMonitor '../../arm/Microsoft.Authorization/policyDefinitions/managementGroup/deploy.bicep' = {
+    name: '${uniqueString(deployment().name)}-rvbuhm-policyDefinitions'
+    params: {
+        name: 'Deploy_RecoveryVault_BackupHealthMonitor_Alert'
+        displayName: '[Modify] Deploy RV Backup Health Monitoring Alerts'
+        description: 'Modify policy to audit/update Recovery Vault Backup Health Alerting to Azure monitor alerts'
+        location: policyLocation
+        metadata: {
+            version: '1.0.0'
+            Category: 'Site Recovery'
```
Comment (jfaurskov) [inline src/resources/Microsoft.Authorization/policyDefinitions/deploy-rv_backuphealth_monitor.bicep:29]:
```suggestion
            category: 'Site Recovery'
```

**Review by jfaurskov — CHANGES_REQUESTED** [review]
Great work @arjenhuitema, just a few comments :-)

File: docs/wiki/Monitoring-and-Alerting-in-ALZ.md (hunk 2 above)
Comment (arjenhuitema) [reply docs/wiki/Monitoring-and-Alerting-in-ALZ.md:113]: This comes directly from the description in the public docs. There are multiple possible configurations aside from "alertsForAllJobFailures", so you can have control over the type of alert that is generated. The only type of alert that you can't disable would be anything that is considered a destructive operation, which should include backup deletion etc. I can´t find all the specific operations that are included in that. 

If you think it´s better, I could rephrase this section.

Comment (jfaurskov) [reply docs/wiki/Monitoring-and-Alerting-in-ALZ.md:113]: No worries, if it's official documentation let's leave it as is.

**Review by jfaurskov — CHANGES_REQUESTED** [review]
Great work @arjenhuitema could you please update the link to MonitoringAndAlerting in the main Readme as well? Then I'll approve :-)

**Comment by arjenhuitema** [comment]
> Great work @arjenhuitema could you please update the link to MonitoringAndAlerting in the main Readme as well? Then I'll approve :-)

Done! Thanks Jan.

**Review by jfaurskov — APPROVED** [review]
LGTM

**Pull request merged**